### ISSUE TITLE
Countables in Node

### ISSUE THREAD

**Issue by onyxnz** [backdrop/backdrop-issues#6134]
## Description of the bug

Switching to user throws nasty error.

## Steps To Reproduce

To reproduce the behavior:
1. Instal Devel
2. Add a user account with elevated privileges that is not admin, eg editor
3. Use the Devel Switch to User to change to that user.

## Actual behavior
TypeError: count(): Argument #1 ($value) must be of type Countable\|array, null given in node_layout_load_by_router_item_alter() (line 1679 of /var/www/clients/client1/web1/web/core/modules/node/node.module).
--

## Expected behavior

Normal usage as elevated user account

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: 1.25
- Web server and its version: nGinx 1.18.0
- PHP version: 8.0,8.1
- Database sever (MySQL or MariaDB?) and its version: 10.6.12-MariaDB-0ubuntu0.22.04.1
- Operating System and its version: Ubu 22.04
- Browser(s) and their versions: Vivaldi/Falkon/Firefox current

**Comment by onyxnz** [comment]
Node.module line 1675-
`function node_layout_load_by_router_item_alter(&$layouts, $router_item) {
  $map = $router_item['map'];
  // Check for node previews.
  if (count($map) == 4 && $map[0] == 'node' && $map[1] == 'preview') {
    // Add any node/% path as a fallback and set the context to the appropriate
    // node from the tempstore.
    $path_is_preview = FALSE;`

There's no checking that $map is containing a countable, which of course PHP8 barfs on.

**Comment by argiepiano** [comment]
I'm attempting to reproduce with PHP 8.1 following your steps. No luck so far. Did you leave something out in the description?

**Comment by onyxnz** [comment]
No, I tested this on a live and a dev site with same codeset, same modules. Perhaps a contrib is contributing to the overloading of the user, such that there is a NULL appearing in the $router_item['map']? 
Of note, I also have Login Destination to send the user to a dashboard.

**Comment by argiepiano** [comment]
Does the user have Dashboard access?

**Comment by onyxnz** [comment]
Yes

**Comment by onyxnz** [comment]
Although I have a custom dashboard in process instead, via a Layout.

**Comment by argiepiano** [comment]
Hmm... installed Login Destination, still no luck. The editor is redirected normally to the core Dashboard. Just saw you have a custom dashboard. Care to share what's in that layout and how you built it?

**Comment by argiepiano** [comment]
I'm starting to suspect you have a NULL router item when `node_layout_load_by_router_item_alter()` is invoked. Did you get a warning in line 1676 of node.module before the nasty error?

**Comment by onyxnz** [comment]
Yes, Warning: Undefined array key "map" in node_layout_load_by_router_item_alter() (line 1676 of /var/www/clients/client1/web1/web/core/modules/node/node.module).
@argiepiano Turning OFF dashboard access, because it's now redundant, fixed this, thanks. 
My custom dashboard uses a custom template/layout/path, with all custom blocks added. Login destination sends the role to the path upon login.

**Comment by argiepiano** [comment]
Hmm, interesting. Can you check `admin/dashboard/settings` to see if the settings for the core Dashboard still redirects users to the core Dashboard?

**Comment by onyxnz** [comment]
Yes, that is still checked, to redirect the Administrators. So the two modules are arm-wrestling.

**Comment by argiepiano** [comment]
> So the two modules are arm-wrestling.

That's what it looks like. So, removing access for the Editor to the core Dashboard has solved this for you?

**Comment by onyxnz** [comment]
Yes, thanks. I still respectfully submit that there should be a check of valid values coming through; eg is_array($router_item['map']) or such. Cheers mate!
